pkg/workload/workload.go

```diff
@@ -258,6 +258,9 @@ func UnsetAdmissionWithCondition(
 		Message:            api.TruncateConditionMessage(message),
 	}
 	newWl := BaseSSAWorkload(wl)
+	// Use resourceVersion to avoid overriding admissions by the scheduler that
+	// happen in a different routine.
+	newWl.ResourceVersion = wl.ResourceVersion
 	newWl.Status.Conditions = []metav1.Condition{condition}
 	return c.Status().Patch(ctx, newWl, client.Apply, client.FieldOwner(constants.AdmissionName))
```

Comment:
Can we avoid conflicts by setting a force option?

According to [docs](https://kubernetes.io/docs/reference/using-api/server-side-apply/#using-server-side-apply-in-a-controller), setting a force option is strongly recommended.

Reply:
Force is only relevant if there was another fieldOwner that overrode the same fields that this call is trying to update.

But we actually don't want to avoid conflicts, we instead want to abort if there are conflicts, because that possibly means that the scheduler admitted the workload in the meantime.